control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -702,7 +703,7 @@ func (r *HostedControlPlaneReconciler) reconcileAPIServerService(ctx context.Con
 	}
 	p := kas.NewKubeAPIServerServiceParams(hcp)
 	apiServerService := manifests.KubeAPIServerService(hcp.Namespace)
-	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, apiServerService, func() error {
+	if _, err := upsert.CreateOrUpdate(ctx, r.Client, apiServerService, func() error {
```

Comment:
I think this would work better as an interface that is injected into the controller in main.go